pkg/codegen/dotnet/gen_program_expressions.go

```diff
@@ -650,6 +719,18 @@ func (g *generator) GenRelativeTraversalExpression(w io.Writer, expr *model.Rela
 	g.genRelativeTraversal(w, expr.Traversal, expr.Parts, nil)
 }
 
+func (g *generator) schemaTypeName(schemaType *schema.ObjectType) string {
+	fullyQualifiedTypeName := schemaType.Token
+	nameParts := strings.Split(fullyQualifiedTypeName, ":")
+	return names.Title(nameParts[len(nameParts)-1])
+}
+
+func (g *generator) withinFunctionInvoke(run func()) {
+	g.insideFunctionInvoke = true
+	run()
+	g.insideFunctionInvoke = false
```

Comment:
As I mentioned for the `withingAsyncBlock`, this should cache the old value, not set to false on exit.

Reply:
Maybe I am bit tired, not sure what you mean. I just did this instead:
```go
func (g *generator) withinAwaitBlock(run func()) {
	if g.insideAwait {
		// already inside await block?
		// only run the function
		run()
	} else {
		// not inside await? flag it as true, run the function,
		// then set it back to false
		g.insideAwait = true
		run()
		g.insideAwait = false
	}
}
```
Even though this doesn't really happen, don't think we double `await`, nor we call function invokes in succession

Reply:
I meant something like:

```go
func (g *generator) withinAwaitBlock(run func()) {
        defer func(v bool) { g.insideAwait = v }(g.insideAwait)
        g.insideAwait = true
        run()
}
```

I think what you put has the same effect.